### PR TITLE
Publish module API documentation generated with doxygen

### PR DESCRIPTION
**Pull request by ligurio**
Generated docs: https://bronevichok.ru/static/module_api/

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Publish
+
+on:
+  push:
+    branches: [master]
+    tags: ['*']
```
Comment (Totktonada) [inline .github/workflows/publish.yaml:6]: It'll run on, say, tagging 1.10.X. I think we should leave only `master` here.

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+      - name: Build module API documentation using Doxygen
+        run: |
+            cmake .
+            make module_api
+            sed -e "s%@PROJECT_.\+_DIR@/%%" Doxyfile.API.in > Doxyfile.API
+            doxygen Doxyfile.API
+
+      - name: Publish generated API documentation to GitHub Pages
+        uses: JamesIves/github-pages-deploy-action@4.1.4
```
Comment (Totktonada) [inline .github/workflows/publish.yaml:26]: There is v4.4.0: https://github.com/JamesIves/github-pages-deploy-action/releases

Worth to fix the latest version?

File: .github/workflows/publish.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: Publish
```
Comment (Totktonada) [inline .github/workflows/publish.yaml:1]: Nit: We have many things to publish (source tarballs, packages), so I would name this workflow file `module-api.yaml` or `publish-module-api.yaml` (and the same in the `name` field inside the file).

File: .github/workflows/publish.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+      - name: Setup Doxygen
+        run: sudo apt install -y doxygen
+
+      - name: Build module API documentation using Doxygen
+        run: |
+            cmake .
+            make module_api
+            sed -e "s%@PROJECT_.\+_DIR@/%%" Doxyfile.API.in > Doxyfile.API
+            doxygen Doxyfile.API
```
Comment (Totktonada) [inline .github/workflows/publish.yaml:23]: AFAIR, doxygen emits a couple of warnings. We can add the similar job into `lint.yaml`, which will not publish anything, but will grep warnings and fail if there is one. The idea is borrowed from [here](https://github.com/tarantool/graphql.0/blob/05f39946299cb2f35a97be326b992aace0205eaf/Makefile#L26-L31).

It is just side idea. Feel free to just close this discussion.

File: .github/workflows/publish.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+      - name: Publish generated API documentation to GitHub Pages
+        uses: JamesIves/github-pages-deploy-action@4.1.4
+        with:
+          branch: gh-pages
+          folder: doc/api/html/
```
Comment (Totktonada) [inline .github/workflows/publish.yaml:29]: Can we generate just one page from one header file using doxygen? Or copy `module_8h.html` to `index.html`?

I looked on the build documentation and it is a bit unexpected that one need to do several click to get the documention: open the page -> click `File list` -> click `module.h`.

File: .github/workflows/publish.yaml (hunk 4 above)
Comment (Totktonada) [inline .github/workflows/publish.yaml:23]: I'm just curious: how about a custom css theme to make the result a bit more nice?

I searched the web across and found those ones:

https://jothepro.github.io/doxygen-awesome-css/ (just css file)
https://mcss.mosra.cz/documentation/doxygen/ (a script to process doxygen generated files)

Feel free to ignore/postpone it.

**Review by Totktonada — APPROVED** [review]
I have no objections. There are several proposals around, but everything can be ignored/postponed.

One important comment: please, drop `tags: ['*']`. LGTM after this.

File: doc/index.html [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<!-- HTML index file used in https://tarantool.github.io/ -->
```
Comment (Totktonada) [inline doc/index.html:1]: Nit: I guess it'll be `https://tarantool.github.io/tarantool` (as I see from other repositories).

File: .github/workflows/publish.yaml (hunk 4 above)
Comment (Totktonada) [reply .github/workflows/publish.yaml]: Okay.

**Comment by Totktonada** [comment]
I looked over the changes and it looks okay. Thank you!

**Review by igormunkin — APPROVED** [review]
@ligurio, thanks for the patch! LGTM, as trivial.

**Review by igormunkin — APPROVED** [review]
@ligurio, thanks for the patch! Nicely done, LGTM.

**Pull request merged**